File: game_server/ServerProtocol/src/Common/QuestEnum/EQuestStatus.java
package Common.QuestEnum;

/*********
 * 任务状态
 **/
public enum EQuestStatus {
	NONE, //0 ==== 
	WAITING, //1 ==== 等待中
	PROGRESSING, //2 ==== 进行中
	;
public static final EQuestStatus[]  EQuestStatus_Values = EQuestStatus.values();
public static final int EQuestStatus_Length = EQuestStatus_Values.length;
public static EQuestStatus EQuestStatus_FromInt(int _ivalue) {
	if(_ivalue < 0 || _ivalue >= EQuestStatus_Length){ return null; }
	return EQuestStatus_Values[_ivalue];
}
}

File: game_client/Assets/Scripts/ClientProtocol/Common/QuestEnum/EQuestStatus.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;
using UnityEngine;


namespace Common.QuestEnum
{

/// <summary>
/// 任务状态
/// </summary>
public enum EQuestStatus {
	NONE, //0 ==== 
	[InspectorName("WAITING - idx[1] - 等待中")]
	WAITING, //1 ==== 等待中
	[InspectorName("PROGRESSING - idx[2] - 进行中")]
	PROGRESSING, //2 ==== 进行中
}

public class EQuestStatusComparer : IEqualityComparer<EQuestStatus>{
	public bool Equals(EQuestStatus x, EQuestStatus y) { return x == y; }
	public int GetHashCode(EQuestStatus obj) { return obj.GetHashCode(); }
	public static int g_iEnumCount = 3;
}
}

